src/message_wrapper.cpp

```diff
@@ -863,14 +937,14 @@ const geometry_msgs::PointStamped MessageWrapper::createRosPointStampedMessage(c
   equatorial_radius = 6378137.0;
   polar_radius      = 6356752.314245;
   eccentricity      = 1 - pow(polar_radius, 2) / pow(equatorial_radius, 2);
-  latitude          = sbgDegToRadD(ref_sbg_ekf_msg.position.x);
-  longitude         = sbgDegToRadD(ref_sbg_ekf_msg.position.x);
+  latitude          = sbgDegToRadD(ref_sbg_ekf_msg.latitude);
+  longitude         = sbgDegToRadD(ref_sbg_ekf_msg.longitude);
 
   prime_vertical_radius = equatorial_radius / sqrt(1.0 - pow(eccentricity, 2) * pow(sin(latitude), 2));
 
-  point_stamped_message.point.x = (prime_vertical_radius + ref_sbg_ekf_msg.position.z) * cos(latitude) * cos(longitude);
-  point_stamped_message.point.y = (prime_vertical_radius + ref_sbg_ekf_msg.position.z) * cos(latitude) * sin(longitude);
-  point_stamped_message.point.z = ((pow(polar_radius, 2) / pow(equatorial_radius, 2)) * prime_vertical_radius + ref_sbg_ekf_msg.position.z) * sin(latitude);
+  point_stamped_message.point.x = (prime_vertical_radius + ref_sbg_ekf_msg.altitude) * cos(latitude) * cos(longitude);
```

Comment:
Probably missing ENU/NED conv here

Reply:
ECEF is the earth-centered, earth-fixed frame. The z axis extends through true north, the x axis intersects the sphere of the earth at latitude=0 and and longitude=0
So it doesn't change